app/code/Magento/Sales/Model/Order/Invoice.php

```diff
@@ -236,7 +236,7 @@ public function setOrder(\Magento\Sales\Model\Order $order)
     public function getOrder()
     {
         if (!$this->_order instanceof \Magento\Sales\Model\Order) {
-            $this->_order = $this->_orderFactory->create()->load($this->getOrderId());
+            $this->_order = $this->orderRepository->get($this->getOrderId());
         }
         return $this->_order->setHistoryEntityName($this->entityType);
```

Comment:
* Method doc block still states that return type is `Magento\Sales\Model\Order` which may not be true in some cases. It's best to change it to `OrderInterface` now. 
* Method clients must be checked to avoid unexpected consequences.
* `setOrder` method of this class is now inconsistent with `getOrder`. It's best to check possibility of aligning signatures and behaviors.